src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/SuspectedTransLocDetector.java

```diff
@@ -0,0 +1,71 @@
+package org.broadinstitute.hellbender.tools.spark.sv.discovery.prototype;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.logging.log4j.Logger;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.*;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SVVCFWriter;
+import org.broadinstitute.hellbender.utils.Utils;
+import scala.Tuple2;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+final class SuspectedTransLocDetector implements VariantDetectorFromLocalAssemblyContigAlignments {
+
+    @SuppressWarnings("unchecked")
+    private static final List<String> EMPTY_INSERTION_MAPPINGS = Collections.EMPTY_LIST;
+
+    @Override
+    public void inferSvAndWriteVCF(final JavaRDD<AlignedContig> localAssemblyContigs, final String vcfOutputFileName,
+                                   final Broadcast<ReferenceMultiSource> broadcastReference,
+                                   final Broadcast<SAMSequenceDictionary> referenceDictionaryBroadcast,
+                                   final Logger toolLogger) {
+        localAssemblyContigs.cache();
```

Comment:
Do you need this `cache` here? Where is it getting unpersisted? I think it would be better to have the calling method handle caching and unpersisting RDDs so that we don't lose track of all the things we've cached.

Reply:
agree and removed.